assets/src/main/java/bisq/asset/coins/CloakCoin.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * This file is part of Bisq.
+ *
+ * Bisq is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.asset.coins;
+
+import bisq.asset.AddressValidationResult;
+import bisq.asset.Base58BitcoinAddressValidator;
+import bisq.asset.Coin;
+import bisq.asset.NetworkParametersAdapter;
+
+public class CloakCoin extends Coin {
+    public CloakCoin() {
+        super("CloakCoin", "CLOAK", new CloakCoinAddressValidator());
+    }
+
+    public static class CloakCoinAddressValidator extends Base58BitcoinAddressValidator {
+
+        public CloakCoinAddressValidator() {
+            super(/*new CloakCoinParams()*/);
+        }
+
+        @Override
+        public AddressValidationResult validate(String address) {
+            if (!address.matches("^[B|C][a-km-zA-HJ-NP-Z1-9]{33}$") && !address.matches("^smY[a-km-zA-HJ-NP-Z1-9]{99}$"))
```

Comment:
This is pure regex validation and does not use anything from super class, so just use `RegexAddressValidator` instead of creating your own validator.